src/Microsoft.AspNetCore.OData/Batch/ODataBatchReaderExtensions.cs

```diff
@@ -236,5 +275,109 @@ private static HttpContext CreateHttpContext(HttpContext originalContext)
 
             return context;
         }
+
+        /// <summary>
+        /// Extract preferences that can be inherited from the overall batch request to
+        /// an individual request.
+        /// </summary>
+        /// <param name="batchPreferences">The value of the Prefer header from the batch request</param>
+        /// <returns>comma-separated preferences that can be passed down to an individual request</returns>
+        private static string GetPreferencesToInheritFromBatch(string batchPreferences)
+        {
+            IEnumerable<string> preferencesToInherit = SplitPreferences(batchPreferences)
+                .Where(value => 
+                    !nonInheritablePreferences.Any(
+                        prefToIgnore => 
+                        value.Trim().ToLowerInvariant().StartsWith(prefToIgnore)
+                    )
+                )
+                .Select(value => value.Trim());
```

Comment:
@habbes Maybe return trimmed values from `SplitPreferences` such that you don't need to call `.Trim()` that many places